plugins/lsp/wtf.nix

```diff
@@ -0,0 +1,122 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.wtf;
+
+  defaultKeymaps = {
+    ai = {
+      key = "gw";
+      mode = ["n" "x"];
+      action = "require('wtf').ai";
+      lua = true;
+    };
+
+    search = {
+      key = "gW";
+      mode = "n";
+      action = "require('wtf').search";
+      lua = true;
+    };
+  };
+in {
+  options = {
+    plugins.wtf =
+      helpers.extraOptionsOptions
+      // {
+        enable = mkEnableOption "wtf.nvim";
+
+        package = helpers.mkPackageOption "wtf.nvim" pkgs.vimPlugins.wtf-nvim;
+
+        keymaps =
+          mapAttrs
+          (
+            action: defaults:
+              helpers.mkNullOrOption
+              (
+                with types;
+                  either
+                  str
+                  (helpers.keymaps.mkMapOptionSubmodule defaults)
+              )
+              "Keymap for the ${action} action."
+          )
+          defaultKeymaps;
+
+        popupType = helpers.defaultNullOpts.mkEnum ["popup" "horizontal" "vertical"] "popup" ''
+          Default AI popup type.
+        '';
+
+        openaiApiKey = helpers.mkNullOrOption types.str "An alternative way to set your API key";
```

Comment:
It would be nice to be either str or raw, to allow using an env var if you don't want a private key in the nix store

Reply:
Good idea ! Done.